clean up console script

dhellmann opened the pull request:
Addresses #3

janbrohl commented on setup.cfg:
```diff
@@ -32,7 +32,7 @@ namespace_packages =
 [entry_points]
 console_scripts =
-    test-template = sphinxcontrib.datatemplates.cli:main
+    sphinx-datatemplate = sphinxcontrib.datatemplates.cli:main
```
While the library depends on sphinx and got sphinx in it's name, the CLI script is not connected to sphinx directly. Just ``datatemplate`` seems better except it does not obviously belongs to ``sphinxcontrib.datatemplate``.

dhellmann replied:
Yes, you make a good point. I was looking for a name that tied it back to this library, because names like "render-template" seemed too generic. I'll leave this open a little longer to give me time to think about a better name.

janbrohl commented on sphinxcontrib/datatemplates/cli.py:
```diff
@@ -38,11 +38,16 @@ def main():
-        k.replace("-", "_"): v
-        for k, _, v in (s.partition(':') for s in args.option)
-    })
+    # Add options. If there is a colon, use it to separate the option
```
Much nicer!

dhellmann commented:
I have changed the command line tool name to `datatemplate` and added a sub-command `render`. I'll add some other sub-commands in other pull requests for things like dumping the data after loading it with various options to help explore how those options are interpreted.

dhellmann commented:
I went ahead and added a `dump` subcommand in this PR.

janbrohl reviewed (approved):
Looks good to me.
